pkg/docker/config/config.go

```diff
@@ -541,6 +542,28 @@ func getPathToAuthWithOS(sys *types.SystemContext, goOS string) (string, bool, e
 // or returns an empty dockerConfigFile data structure if auth.json does not exist
 // if the file exists and is empty, readJSONFile returns an error
 func readJSONFile(path string, legacyFormat bool) (dockerConfigFile, error) {
+	if _, err := os.Stat(path); err != nil {
+		if os.IsNotExist(err) {
+			var auths dockerConfigFile
+			auths.AuthConfigs = map[string]dockerAuthConfig{}
+			return auths, nil
+		}
+		return dockerConfigFile{}, err
+	}
+
+	// Make sure to read-lock the file before reading it.
+	lock, err := lockfile.GetLockfile(path)
+	if err != nil {
+		return dockerConfigFile{}, fmt.Errorf("creating lock: %w", err)
+	}
+	lock.RLock()
+	defer lock.Unlock()
+
+	return readJSONFileNoLock(path, legacyFormat)
+}
+
+// Implements readJSONFile but without locking the specified path.
+func readJSONFileNoLock(path string, legacyFormat bool) (dockerConfigFile, error) {
```

Comment:
Something like
```suggestion
// Implements readJSONFile but without locking the specified path. The caller is responsible for locking the lock.
func readJSONFileLocked(path string, legacyFormat bool) (dockerConfigFile, error) {
```
to make it very clear that it’s not OK to just call this method without holding a lock?

At least the `…Locked` naming convention is used in other places in c/image.